crates/primitives/src/signature/sig.rs

```diff
@@ -1,821 +1,144 @@
-        Self::from_rs_and_parity(r, s, parity)
-            .map_err(|_| alloy_rlp::Error::Custom("attempted to decode invalid field element"))
-    }
-}
+    #[cfg(feature = "rlp")]
+    fn decode_rlp_vrs(buf: &mut &[u8]) -> Result<Self, alloy_rlp::Error>
+    where
+        Self: Sized;
 
-impl Signature {
     #[doc(hidden)]
-    pub fn test_signature() -> Self {
-        Self::from_scalars_and_parity(
-            b256!("840cfc572845f5786e702984c2a582528cad4b49b2a10b9db1be7fca90058565"),
-            b256!("25e7109ceb98168d95b09b18bbf6b685130e0562f233877d492b94eee0c5b6d1"),
-            false,
-        )
-        .unwrap()
-    }
-
-    /// Instantiate a new signature from `r`, `s`, and `v` values.
-    #[allow(clippy::missing_const_for_fn)]
-    pub fn new(r: U256, s: U256, v: Parity) -> Self {
-        Self { r, s, v }
-    }
-
-    /// Returns the inner ECDSA signature.
-    #[cfg(feature = "k256")]
-    #[deprecated(note = "use `Signature::to_k256` instead")]
-    #[inline]
-    pub fn into_inner(self) -> k256::ecdsa::Signature {
-        self.try_into().expect("signature conversion failed")
-    }
-
-    /// Returns the inner ECDSA signature.
-    #[cfg(feature = "k256")]
-    #[inline]
-    pub fn to_k256(self) -> Result<k256::ecdsa::Signature, k256::ecdsa::Error> {
-        k256::ecdsa::Signature::from_scalars(self.r.to_be_bytes(), self.s.to_be_bytes())
-    }
-
-    /// Instantiate from a signature and recovery id
-    #[cfg(feature = "k256")]
-    pub fn from_signature_and_parity<T: TryInto<Parity, Error = E>, E: Into<SignatureError>>(
-        sig: k256::ecdsa::Signature,
-        parity: T,
-    ) -> Result<Self, SignatureError> {
-        let r = U256::from_be_slice(sig.r().to_bytes().as_ref());
-        let s = U256::from_be_slice(sig.s().to_bytes().as_ref());
-        Ok(Self { v: parity.try_into().map_err(Into::into)?, r, s })
-    }
-
-    /// Creates a [`Signature`] from the serialized `r` and `s` scalar values, which comprise the
-    /// ECDSA signature, alongside a `v` value, used to determine the recovery ID.
-    #[inline]
-    pub fn from_scalars_and_parity<T: TryInto<Parity, Error = E>, E: Into<SignatureError>>(
-        r: crate::B256,
-        s: crate::B256,
-        parity: T,
-    ) -> Result<Self, SignatureError> {
-        Self::from_rs_and_parity(
-            U256::from_be_slice(r.as_ref()),
-            U256::from_be_slice(s.as_ref()),
-            parity,
-        )
-    }
-
-    /// Normalizes the signature into "low S" form as described in
-    /// [BIP 0062: Dealing with Malleability][1].
-    ///
-    /// [1]: https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki
-    #[inline]
-    pub fn normalize_s(&self) -> Option<Self> {
-        let s = self.s();
-
-        if s > SECP256K1N_ORDER >> 1 {
-            Some(Self { v: self.v.inverted(), r: self.r, s: SECP256K1N_ORDER - s })
-        } else {
-            None
-        }
-    }
+    fn test_signature() -> Self
+    where
+        Self: Sized;
```

Comment:
this shouldn't be in the trait imo. if it is, then with attribute `#[cfg(test)]` or `#[cfg(feature = test-util]`, but that would change the logic here, but this is a refactor pr, so leave it out, just move it to the `EcdsaSignature` impl body.